server/metadata/dictionary.go

```diff
@@ -153,23 +178,27 @@ func (r *reservedSubspace) reserveNamespace(ctx context.Context, tx transaction.
 
 	r.RLock()
 	defer r.RUnlock()
-	if allocatedTo, ok := r.allocated[id]; ok {
-		log.Debug().Uint32("namespace_id", id).Str("namespace_name", r.allocated[id]).Msg("namespace reserved for")
-		if allocatedTo == namespace {
-			return nil
-		} else {
-			return errors.AlreadyExists("id is already assigned to the namespace '%s'", allocatedTo)
+	if _, ok := r.idToNamespaceStruct[namespaceMetadata.Id]; ok {
```

Comment:
add a `namespaceMetadata.Id` zero check above this as well i.e. if the `id` is zero and passed to this method then we reject.

Reply:
fixed.